core/dbt/contracts/project.py

```diff
@@ -180,6 +180,13 @@ class RegistryPackageMetadata(
 }
 
 
+@dataclass
+class SchemaManagementConfiguration(HyphenatedDbtClassMixin, Replaceable):
+    database: Optional[str] = None
+    schema: Optional[str] = None
```

Comment:
Big question: **Do schemas feel like the right level of abstraction?**

Is this the raw `schema` config that users are setting in dbt projects, or the final `schema` that takes into account env-specific logic (`generate_schema_name` macro)?

Say, on Snowflake, where each user could theoretically have their own logical database. There might be entire schemas that have become unused cruft, and need managing (warning/dropping). If the user has to manually specify all of those schemas every time, that doesn't feel scalable. Maybe they should be able to specify an entire `database` for dbt to manage, opening up the possibility that dbt would drop entire schemas in addition to dropping specific relations?

Reply:
Alternative idea (possibly a bad one): Should dbt be introspecting database permissions, to determine the set of schemas/objects for which it is `owner`?